run_all_experiments.py

```diff
@@ -50,11 +50,13 @@
 BENCHMARK_DIR: str = f'{BENCHMARK_ROOT}/comparison'
 RESULTS_DIR: str = run_one_experiment.RESULTS_DIR
 GENERATED_BENCHMARK: str = 'generated-benchmark-'
+PROJECT_SUMMARY_JSON = os.path.join(RESULTS_DIR, 'project_coverage_gain.json')
```

Comment:
Have you thought about having an actual experiment summary file instead of one specific to coverage gain. Right now there are a few pieces of information that we infer from the file path but would be good to have in a summary file, e.g. project name, function name, function signature, etc.

Reply:
Dongge opened #105 a while ago to add a information.json, it would be good to integrate this into that instead of creating another file.

Reply:
This is an interesting point; thanks for bringing it up. Some thoughts:
1. In #105, I considered having 1 file per benchmark to avoid concurrent reading/writing to the same file. We need a lock if we want a single file for all (like the following) because multiple benchmarks may modify the same file.
```json
{
  "project_A": {
    "benchmarks": {
      "benchmark_dir_Aa": {"info_1": "content_1Aa", "info_2": "content_2Aa", ...},
      "benchmark_dir_Ab": {"info_1": "content_1Ab", "info_2": "content_2Ab", ...},
    },
    "coverage_gainA": "valueA",
    ...,
  },
  "project_B": {
    "benchmarks": {
      "benchmark_dir_Ba": {"info_1": "content_1Ba", "info_2": "content_2Ba", ...},
      "benchmark_dir_Bb": {"info_1": "content_1Bb", "info_2": "content_2Bb", ...},
    },
    "coverage_gainB": "valueB",
    ..., 
  },
}
```

2. Given adding other info in 105 will require many modifications, how about:
    1. In this PR, we design the structure of this file to be expandable to add the other infos in 105.
    2. In a later PR, we store those infos in 105.

Any other thoughts?